server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -627,12 +627,12 @@ public String apply(Sink input)
 
         synchronized (handoffCondition) {
           while (!sinks.isEmpty()) {
-            handoffCondition.wait();
+            handoffCondition.wait(config.getHandoffConditionTimeout());
```

Comment:
`sinks.isEmpty` will remain false after this wait times out, so this wait will just keep getting called. Wait can also spuriously wake up. So this should compute how long to wait based on when we started waiting, and then the "while" should check whether or not we are done waiting yet and maybe call wait again or maybe give up.